tests/test_aliases.py

```diff
@@ -702,3 +702,26 @@ class Foo(BaseModel):
     assert f.a == 'a'
     assert f.model_dump(by_alias=True) == {'a_ser_alias': 'a'}
     assert f.model_dump(by_alias=False) == {'a': 'a'}
+
+
+def test_alias_generator_with_computed_field_for_serialization() -> None:
+    """Tests that the alias generator is used for computed fields, with serialization_alias taking precedence over alias."""
+
+    class Rectangle(BaseModel):
+        model_config = ConfigDict(
+            alias_generator=AliasGenerator(
+                validation_alias=lambda field_name: f'{field_name}_val_alias',
+                alias=lambda field_name: f'{field_name}_alias',
+                serialization_alias=lambda field_name: f'{field_name}_ser_alias',
+            )
```

Comment:
Could we add a similar test with `serialization_alias=None` or whatever? And make sure it does the right thing using just the `alias`

Reply:
That test already exists:

```py
@pytest.mark.parametrize('alias_generator', upper_alias_generator)
def test_alias_generator_with_computed_field(alias_generator) -> None:
    class Rectangle(BaseModel):
        model_config = ConfigDict(populate_by_name=True, alias_generator=alias_generator)

        width: int
        height: int

        @computed_field
        @property
        def area(self) -> int:
            return self.width * self.height

    r = Rectangle(width=10, height=20)
    assert r.model_dump(by_alias=True) == {'WIDTH': 10, 'HEIGHT': 20, 'AREA': 200}
```